backend/src/rss-feed/rss-feed.service.ts

```diff
@@ -25,8 +26,28 @@ export class RssFeedService {
     @Transaction()
     async fetchAndSaveRssFeeds() {
         try {
-            const [news, stories] = await Promise.all([parser.parseURL(sources.news), parser.parseURL(sources.stories)]);
-            await Promise.all([this.rssFeedRepository.save(news.items), this.rssFeedRepository.save(stories.items)]);
+            const [
+                news,
```

Comment:
You are fetching data from all sources in parallel
But you are waiting until all the sources are fetched to begin putting them to database ?

Why don't you just fetch individual pieces and start saving them individually right after each piece are fetched?

Code example are:
```
const addSourceName = name => {  return (data) => {/*... code to injectName */ return transformedData} }
const persistToDb = data => this.rssFeedRepository.save(data)
 parseURL(datasource).then(addSourceName('sourceName')).then(persistIntoDb)
 /* repeat saving */
```